### PR TITLE
Faction editor update

### PR DESCRIPTION
**Pull request by Druss99**
This PR allows users to edit a faction and save it as a new faction. The two improvements I am unsure how to implement is:
- Currently when a checkbox is unticked, adding a unit will retick the checkbox, probably need a way to run through the checkboxes, see what is checked, and recreate them. Or just change it to not recreate all check boxes and instead just add a checkbox for the unit added
- Little python boxes pop up (presumably due to the _format method) when the new game wizard is launched

### CODE REVIEW

File: qt_ui/windows/newgame/WizardPages/QFactionSelection.py [hunk 1]
```diff
@@ -45,12 +55,13 @@ def _add_checkboxes(
         combo_layout: Optional[QHBoxLayout] = None,
     ) -> int:
         counter += 1
-        for i, v in enumerate(sorted(units, key=lambda x: str(x)), counter):
-            cb = QCheckBox(str(v))
-            cb.setCheckState(Qt.CheckState.Checked)
-            self.checkboxes[str(v)] = cb
-            grid.addWidget(cb, i, 1)
-            counter += 1
+        if len(units) > 0:
```
Comment (Raffson) [inline qt_ui/windows/newgame/WizardPages/QFactionSelection.py]: Any particular reason for this if-statement? Doesn't look like it provides any added value 🤔

File: qt_ui/windows/newgame/WizardPages/QFactionSelection.py [hunk 2]
```diff
@@ -228,6 +321,22 @@ def updateFactionUnits(self, units: Union[set, list]):
...
+        hbox = QHBoxLayout()
+        hbox.addWidget(cb)
+        hbox.addWidget(add_button)
+        if cb.count() == 0:
+            cb.hide()
+            add_button.hide()
+        else:
+            cb.show()
+            add_button.show()
+        return hbox
```
Comment (Raffson) [inline qt_ui/windows/newgame/WizardPages/QFactionSelection.py:337]: These are causing the popups, changing them to `cb.setEnabled(False)` or `cb.setEnabled(True)` (and for `add_button` respectively) should fix the issue. If you really want to hide them, you'll need to use `setStyleSheet` and use CSS instead ;)

File: qt_ui/windows/newgame/WizardPages/QFactionSelection.py [hunk 3]
```diff
@@ -284,6 +393,17 @@ def __init__(self, parent=None):
             if r == "USA 2005":
                 self.blueFactionSelect.setCurrentIndex(i)
 
+        self.saveBlueFactionButton = QtWidgets.QPushButton("Save as new faction")
+        self.saveRedFactionButton = QtWidgets.QPushButton("Save as new faction")
```
Comment (Raffson) [inline qt_ui/windows/newgame/WizardPages/QFactionSelection.py]: No need to mention `QtWidgets`, `QPushButton` is already imported...

File: qt_ui/windows/newgame/WizardPages/QFactionSelection.py [hunk 4]
```diff
@@ -380,10 +500,68 @@ def _filter_selected_units(qfu: QFactionUnits) -> Faction:
         qfu.updateFactionUnits(fac.missiles)
         return fac
 
+    def show_save_faction_dialog(self, faction: Faction):
+        dialog = QFactionSaver(faction)
+        dialog.exec_()
```
Comment (Raffson) [inline qt_ui/windows/newgame/WizardPages/QFactionSelection.py]: Prefer `exec` instead, `exec_` is an old thing because `exec` is also a Python function...

**Review by Raffson — CHANGES_REQUESTED** [review]

**Comment by Druss99** [comment]
All errors fixed, will potentially look to refactor some of this longterm but I'd rather ship in 1.4 and afterwards see if the juice is worth the squeeze on a refactor

**Review by Raffson — APPROVED** [review]

**Pull request merged**